src/compiler/builderState.ts

```diff
@@ -76,20 +78,78 @@ namespace ts {
             signature: string | undefined;
             affectsGlobalScope: boolean | undefined;
         }
-        /**
-         * Referenced files with values for the keys as referenced file's path to be true
-         */
-        export type ReferencedSet = ReadonlySet<Path>;
+
+        export interface TwoWayMap<K, V> extends ESMap<K, ReadonlySet<V>> {
+            getKeys(v: V): ReadonlySet<K> | undefined;
+            clone(): TwoWayMap<K, V>;
+        }
+
+        export function createTwoWayMap<K, V>(): TwoWayMap<K, V> {
+            function create(forward: ESMap<K, ReadonlySet<V>>, reverse: ESMap<V, Set<K>>): TwoWayMap<K, V> {

```

Comment:
I think you should pull this function out and name it createTwoWayMapWithForwardAndReverse or something so that its not in closure

Reply:
I'm not sure I see the advantage, since it's not closing over anything.